cmf-cli/Commands/build/business/ValidateStartEndMethods/SolutionValidator.cs

```diff
@@ -0,0 +1,161 @@
+using Cmf.CLI.Commands.build.business.ValidateStartEndMethods.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Cmf.CLI.Commands.build.business.ValidateStartEndMethods;
+
+internal class SolutionValidator
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly string _solutionPath;
+    private readonly IList<string> _files;
+
+    public SolutionValidator(IServiceProvider serviceProvider, string solutionPath, IEnumerable<string> files)
+    {
+        _serviceProvider = serviceProvider;
+        _solutionPath = solutionPath;
+        _files = files.ToList();
+    }
+
+    public async Task ValidateSolution()
+    {
+        var solutionLoader = new SolutionLoader(_solutionPath);
+        var workspace = solutionLoader.ReLoadSolution();
+        var projects = workspace.CurrentSolution.Projects;
+
+        foreach (var project in projects)
+        {
+            foreach (var document in project.Documents)
+            {
+                if (!_files.Any() || _files.Contains(document.Name))
+                {
+                    var syntaxTree = await document.GetSyntaxTreeAsync();
+
+                    if (syntaxTree is null)
+                    {
+                        continue;
+                    }
+
+                    var namespaceNodes = syntaxTree.GetRoot().DescendantNodes().OfType<NamespaceDeclarationSyntax>();
+
+                    foreach (var namespaceNode in namespaceNodes)
+                    {
+                        var classDeclarations = namespaceNode.ChildNodes().OfType<ClassDeclarationSyntax>();
+
+                        foreach (var classDeclaration in classDeclarations)
+                        {
+                            var methodNodes = classDeclaration.ChildNodes().OfType<MethodDeclarationSyntax>();
+                            var classIdentifierToken = classDeclaration.Identifier;
+                            var classType = classDeclaration.GetClassType();
+
+                            if (classType != ClassType.Other)
+                            {
+                                foreach (MethodDeclarationSyntax methodNode in methodNodes)
+                                {
+                                    ProcessMethod(methodNode, namespaceNode, classType, classIdentifierToken);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public void ProcessMethod(MethodDeclarationSyntax methodNode, NamespaceDeclarationSyntax namespaceNode, ClassType classType, SyntaxToken classIdentifierToken)
+    {
+        var block = methodNode.Body;
+        var methodName = methodNode.Identifier.ValueText;
```

Comment:
Maybe move the variables declaration below the "block is null" statement? 
To avoid unnecessary variable declarations.

Reply:
Well thought.